File: app/src/main/java/com/hustunique/morii/ui/ArcImageView.java
package com.hustunique.morii.ui;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Path;
import android.util.AttributeSet;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatImageView;

/**
 * 弧形的view
 */
public class ArcImageView extends AppCompatImageView {
    /*
     *弧形高度
     */
    private final int mArcHeight = 100;

    public ArcImageView(Context context) {
        this(context, null);
    }

    public ArcImageView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs, 0);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        Path path = new Path();
        path.moveTo(0, 0);
        path.lineTo(0, getHeight() - mArcHeight);
        path.quadTo(getWidth() / 2, getHeight() + mArcHeight, getWidth(), getHeight() - mArcHeight);
        path.lineTo(getWidth(), 0);
        path.close();
        canvas.clipPath(path);
        super.onDraw(canvas);
    }
}

File: app/src/main/java/com/hustunique/morii/ui/ArcImageView.kt
package com.hustunique.morii.ui

import android.content.Context
import android.graphics.Canvas
import android.graphics.Path
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView

/**
 * 弧形的view
 */
class ArcImageView @JvmOverloads constructor(context: Context?, attrs: AttributeSet? = null) :
    AppCompatImageView(
        context!!, attrs, 0
    ) {
    /*
     *弧形高度
     */
    private val mArcHeight = 100
    override fun onDraw(canvas: Canvas) {
        val path = Path()
        path.moveTo(0f, 0f)
        path.lineTo(0f, (height - mArcHeight).toFloat())
        path.quadTo(
            (width / 2).toFloat(),
            (height + mArcHeight).toFloat(),
            width.toFloat(),
            (height - mArcHeight).toFloat()
        )
        path.lineTo(width.toFloat(), 0f)
        path.close()
        canvas.clipPath(path)
        super.onDraw(canvas)
    }
}